test/lib/processors/manifestCreator.js

```diff
@@ -422,6 +422,38 @@ test.serial("default manifest creation with special characters small app descrip
 	t.is(await result.getString(), expectedManifestContentSmallVersionString, "Correct result returned");
 });
 
+test.serial("default manifest creation with special characters very small app descriptor version", async (t) => {
+	const {manifestCreator, errorLogStub} = t.context;
+	const prefix = "/resources/sap/ui/mine/";
+	const libraryResource = {
+		getPath: () => {
+			return prefix + ".library";
+		},
+		getString: async () => {
+			return libraryContent;
+		},
+		_project: {
+			dependencies: [{
+				metadata: {
+					name: "sap.ui.core"
+				}
+			}]
+		}
+	};
+	t.is(errorLogStub.callCount, 0);
+
+	const options = {descriptorVersion: new Version("1.1.0")};
```

Comment:
I wonder why the API requires a semver version instance instead of a plain string.
AFAICS the whole processor is not marked as public API so we should be able to do a cleanup.

As this is not related to the actual PR of fixing the `embeds` behavior, I'd be in favor of creating a separate PR.
Or is there a relation from the version handling change (toString) to the embeds change? At least I could not find any hint. Maybe to increase code coverage?

Reply:
inside the manifestCreator the SemVer object is used for version comparison, as api parameter (descriptorVersion) a string definitely makes sense.

Reply:
Also see https://github.com/SAP/ui5-builder/pull/556 where we already discussed making the manifestCreator public and that we would re-check the API before doing so.